tools/SetupFlow/DevHome.SetupFlow/ViewModels/CloneRepoSummaryInformationFileViewModel.cs

```diff
@@ -0,0 +1,229 @@
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(CloneRepoSummaryInformationViewModel));
+
+    private const string _eventName = "CloneRepo_NextSteps_Event";
+
+    private const string _runOperation = "run";
+
+    private const string _viewOperation = "view";
+
+    private readonly Guid _relatedActivityId;
+
+    private readonly ISetupFlowStringResource _stringResource;
+
+    public string FileName => Path.GetFileName(FilePathAndName) ?? string.Empty;
+
+    public bool HasContent => !string.IsNullOrEmpty(FilePathAndName)
+        && !string.IsNullOrEmpty(RepoName)
+        && !string.IsNullOrEmpty(OwningAccount);
+
+    public string FilePathAndName { get; set; } = string.Empty;
+
+    public string RepoName { get; set; } = string.Empty;
+
+    public string OwningAccount { get; set; } = string.Empty;
+
+    // Using the resource file for properties like .Text and .Content does not work in this case because
+    // the UserControl is inside a DataTemplate and does not have access to the string resource file.
+    // any x:Uid used is blank in the view.
+    // Set the strings here.
+    public string FileFoundMessage { get; } = string.Empty;
+
+    public string FileDescription
+    {
+        get
+        {
+            var repoPath = Path.Join(OwningAccount, RepoName);
+            return _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsDescription, FileName, repoPath);
+        }
+    }
+
+    public string ViewFileMessage { get; } = string.Empty;
+
+    public string RunFileMessage { get; } = string.Empty;
+
+    public CloneRepoSummaryInformationViewModel(SetupFlowOrchestrator setupFlowOrchestrator, ISetupFlowStringResource stringResource)
+    {
+        _relatedActivityId = setupFlowOrchestrator.ActivityId;
+        _stringResource = stringResource;
+        FileFoundMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsFileFound);
+        ViewFileMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsView);
+        RunFileMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsRun);
+    }
+
+    [RelayCommand]
+    public void OpenFileInExplorer()
+    {
+        TelemetryFactory.Get<ITelemetry>().Log(
+            _eventName,
+            LogLevel.Critical,
+            new CloneRepoNextStepEvent(_viewOperation, RepoName),
+            _relatedActivityId);
+
+        if (FilePathAndName is null)
+        {
+            TelemetryFactory.Get<ITelemetry>().Log(
+                _eventName,
+                LogLevel.Critical,
+                new CloneRepoNextStepError(_viewOperation, new ArgumentNullException(nameof(FilePathAndName)).ToString(), RepoName),
+                _relatedActivityId);
+
+            _log.Warning("CloneRepoSummaryInformationViewModel", $"{nameof(FilePathAndName)} is null when trying to view file.");
+            return;

```

Comment:
>return; [](http://example.com/codeflow?start=12&length=7)

why do we return success?  how does the caller know it failed?

Reply:
True.  Nothing changes if this fails.  #2487 to track this work.